src/app/pages/deposits/deposits.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Deposit } from 'src/app/models/Deposit';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Subscription } from 'rxjs';
+
+import { DepositsService } from 'src/app/services/deposits/deposits.service';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-deposits',
+  templateUrl: './deposits.component.html',
+  styleUrls: ['./deposits.component.scss']
+})
+export class DepositsComponent implements OnInit, OnDestroy {
+  subscribe: Subscription[] = [];
+  headElements = ['Reference', 'Date', 'Amount'];
+  transactions: Deposit[] = [];
+  startDateVal: string = '';
+  endDateVal: string = '';
+  newResults: any[];
+
+  constructor(
+    private depositsService: DepositsService,
+    private spinner: NgxSpinnerService,
+    private toastrService: ToastrService
+  ) {}
+
+  onFilter() {
+    const filteredTransactions = [];
+    this.transactions.filter(transaction => {
+      const date = new Date(transaction['created_at']);
```

Comment:
There's also a slight `ts-lint` issue here. `object access via string literals is disallowed`

I thought this could help `transaction.created_at`? but I'm getting another `ts-lint` error that says `created_at` is not `defined` on type deposit. Maybe you could add that in the model file

Reply:
Thanks @Tevinthuku , I have worked on this!